src/main/java/org/jenkinsci/maven/plugins/hpi/TestInsertionMojo.java

```diff
@@ -102,22 +105,29 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             getLog().info("Skipping auto-test generation because we are targeting Jenkins "+target+" (at least 1.327 is required).");
             return;
         }
+        if (javaLevel != null && javaLevel.compareTo("7") < 0) {
+            getLog().info("Skipping auto-test generation because java.level= "+javaLevel+" in pom.xml (at least 7 is required).");
+            return;
+        }
 
         try {
             File f = new File(project.getBasedir(), "target/generated-test-sources/injected");
             f.mkdirs();
             File javaFile = new File(f, injectedTestName + ".java");
             PrintWriter w = new PrintWriter(new OutputStreamWriter(new FileOutputStream(javaFile)));
-            w.println("import java.util.*;");
             w.println("/**");
             w.println(" * Entry point to auto-generated tests (generated by maven-hpi-plugin).");
-            w.println(" * If this fails to compile, you are probably using Hudson &lt; 1.327. If so, disable");
-            w.println(" * this code generation by configuring maven-hpi-plugin to &lt;disabledTestInjection&gt;true&lt;/disabledTestInjection&gt;.");
+            w.println(" * It's possible to disable this code generation by configuring maven-hpi-plugin to {@code <disabledTestInjection>true</disabledTestInjection>}.");
```

Comment:
Why are we even explaining this, given what we know about the insane side effects of this?

Reply:
@daniel-beck 
I don't know the answer, but it may be in another PR. (I am eventually trying to make this PR merged since January.)